### ISSUE TITLE
Batch Norm

### ISSUE THREAD

**Issue by nathanielsimard** [burn-rs/burn#154]
### Feature description

Add BatchNorm layer
